pennylane/qchem/vibrational/bosonic_mapping.py

```diff
@@ -0,0 +1,160 @@
+    r"""Returns expression to convert qubit-local term ::math::``\ket{x_j}\bra{x^{'}}``
+    to qubit operators as given in Eq. (6-9) <https://www.nature.com/articles/s41534-020-0278-0>"""
+
+    c1, c2 = 0.5, -0.5 if i == 1 else 0.5
+
+    if i != j:
+        return PauliSentence({PauliWord({qub_id: "X"}): c1, PauliWord({qub_id: "Y"}): c2 * 1j})
+
+    return PauliSentence({PauliWord({}): c1, PauliWord({qub_id: "Z"}): c2})
+
+
+def binary_mapping(
+    bose_operator: Union[BoseWord, BoseSentence],
+    nstates: int = 2,
+    ps: bool = False,
+    wire_map: dict = None,
+    tol: float = None,
+):
+    r"""Convert a bosonic operator to a qubit operator using the standard-binary mapping.
+
+    The mapping procedure is described in `arXiv:1507.03271 <https://arxiv.org/pdf/1507.03271>`_.
+
+    Args:
+        bose_operator(BoseWord, BoseSentence): the bosonic operator
+        nstates(int): maximum number of states a boson can occupy
+        ps (bool): whether to return the result as a PauliSentence instead of an
+            operator. Defaults to False.
+        wire_map (dict): a dictionary defining how to map the states of
+            the Bose operator to qubit wires. If None, integers used to
+            label the bosonic states will be used as wire labels. Defaults to None.
+        tol (float): tolerance for discarding the imaginary part of the coefficients
+
+    Returns:
+        a linear combination of qubit operators
+
+    **Example**
+
+    >>> w = qml.qchem.BoseWord({(0, 0): "+"})
+    >>> qml.qchem.binary_mapping(w, nstates=4)
+    0.6830127018922193 * X(0)
+    + -0.1830127018922193 * X(0) @ Z(1)
+    + -0.6830127018922193j * Y(0)
+    + 0.1830127018922193j * Y(0) @ Z(1)
+    + 0.3535533905932738 * X(0) @ X(1)
+    + -0.3535533905932738j * X(0) @ Y(1)
+    + 0.3535533905932738j * Y(0) @ X(1)
+    + (0.3535533905932738+0j) * Y(0) @ Y(1)
+    """
+
+    qubit_operator = _binary_mapping_dispatch(bose_operator, nstates, tol=tol)
+
+    wires = list(bose_operator.wires) or [0]
+    identity_wire = wires[0]
+    if not ps:
+        qubit_operator = qubit_operator.operation(wire_order=[identity_wire])
+
+    if wire_map:
+        return qubit_operator.map_wires(wire_map)
+
+    return qubit_operator
+
+
+@singledispatch
+def _binary_mapping_dispatch(bose_operator, nstates, tol):
+    """Dispatches to appropriate function if bose_operator is a BoseWord or BoseSentence."""
+    raise ValueError(f"bose_operator must be a BoseWord or BoseSentence, got: {bose_operator}")
+
+
+@_binary_mapping_dispatch.register
+def _(bose_operator: BoseWord, nstates, tol=None):
+
+    if nstates < 2:
+        raise ValueError(
+            f"Number of states a boson can occupy cannot be less than 2, provided {nstates}."
+        )
+    nqub_per_boson = int(np.ceil(np.log2(nstates)))
+
+    cr_op = np.zeros((nstates, nstates))
+    for s in range(nstates - 1):
+        cr_op[s + 1, s] = np.sqrt(s + 1.0)
+
+    op_mat = {"+": cr_op, "-": cr_op.T}
```

Comment:
Seems like we get the coefficients from this `op_mat`. Maybe there is a better name?

Reply:
The naming here is a confusing. `cr_op` is an array of coefficients on an off-diagonal. And `op_mat` is just those coefficients assigned to a sign. But the names don't suggest this, so one needs to carefully read the line `coeff = op_mat[sign][i][j]` to understand that it's the coeffs in there. Also, what are the equations that help us arrive at these coefficients? This still eludes me.

Reply:
You get the coefficients from equation 28 and 29, look at the eigenvalues or you can also check the matrix in equation 27 in the nature paper, that is for q, which is sum of a and a^ operators.

Reply:
changed the name to coeff_mat, I hope that is more clear.